### PR TITLE
Dockerfile for self-supervised examples

### PR DESCRIPTION
**Pull request by Nimrais**
### Pull Request FAQ
- [documentation](https://catalyst-team.github.io/catalyst/)
- [contribution guide](https://github.com/catalyst-team/catalyst/blob/master/CONTRIBUTING.md)
- [minimal examples section](https://github.com/catalyst-team/catalyst#minimal-examples)
- [changelog](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md) for main framework updates
- [Catalyst slack (#__questions channel)](https://join.slack.com/t/catalyst-team-core/shared_invite/zt-d9miirnn-z86oKDzFMKlMG4fgFdZafw) for issue discussion

## Description

We have a lot of self-supervised algorithms to test.
The Docker is the selected option for the testing. 

## Type of Change

- [x] Examples / docs / tutorials / contributors update
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves an existing feature)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

### Checklist
- [ ] Have you updated tests for the new functionality?
- [ ] Have you added your new classes/functions to the docs?
- [ ] Have you updated the [CHANGELOG](https://github.com/catalyst-team/catalyst/blob/master/CHANGELOG.md)?
- [ ] Have you run [colab minimal CI/CD](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/colab_ci_cd.ipynb) with `latest` and `minimal` requirements?
- [ ] Have you checked XLA integration with [single](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/Catalyst_XLA_single_process.ipynb) and [multiple](https://colab.research.google.com/github/catalyst-team/catalyst/blob/master/examples/notebooks/Catalyst_XLA_multi_process.ipynb) processes?

### CODE REVIEW

File: examples/self_supervised/common.py [hunk 1]
```diff
@@ -74,3 +82,52 @@ def forward(self, x):
         emb = self.encoder(x)
         projection = self.model(emb)
         return emb, projection
+
+
+def get_loaders(args) -> Dict[str, DataLoader]:
```
Comment (Scitator) [inline examples/self_supervised/common.py]: let's unpack args here (very bad practice, hard to understand)

File: examples/self_supervised/common.py [hunk 2]
```diff
@@ -74,3 +82,52 @@ def forward(self, x):
...
+    )
+
+    train_loader = DataLoader(train_data, batch_size=args.batch_size, num_workers=args.num_workers)
+
+    valid_loader = DataLoader(valid_data, batch_size=args.batch_size, num_workers=args.num_workers)
+
+    return {"train": train_loader, "valid": valid_loader}
+
+
+def get_contrastive_model(args) -> ContrastiveModel:
```
Comment (Scitator) [inline examples/self_supervised/common.py]: and here ;)

File: examples/self_supervised/common.py (hunk 1 above)
Comment (Scitator) [reply examples/self_supervised/common.py]: dataset, batch_size, num_workers, etc

File: examples/self_supervised/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,6 @@
+FROM python:3.8
+
+RUN pip install catalyst[cv]==21.09
+RUN pip install catalyst[ml]==21.09
```
Comment (Scitator) [inline examples/self_supervised/Dockerfile:4]:
```suggestion
RUN pip install catalyst[cv]==21.09 catalyst[ml]==21.09
```
is it possible to do so?

**Review by Scitator — DISMISSED** [review]

File: examples/self_supervised/common.py (hunk 2 above)
Comment (Nimrais) [reply examples/self_supervised/common.py]: [Done](https://github.com/catalyst-team/catalyst/pull/1322/commits/59f095bd1d1531834098d20e0cc68219604a082c)

File: examples/self_supervised/common.py (hunk 1 above)
Comment (Nimrais) [reply examples/self_supervised/common.py]: [Done](https://github.com/catalyst-team/catalyst/pull/1322/commits/2fb49ab574a844c2c67ab16d2befde885864f21e)

**Pull request merged**